ledger/db/src/ledger_trait.rs

```diff
@@ -8,14 +8,16 @@ use mc_transaction_core::{
     tx::{TxOut, TxOutMembershipProof},
     Block, BlockContents, BlockSignature,
 };
+use mockall::*;
 
-pub trait Ledger: Clone + Send {
```

Comment:
automock cannot mock traits that inherit from Clone.

Reply:
so the pattern here is to switch to using an Arc to wrap things instead of cloning? this seems like it will always be an acceptable tradeoff for us to have mocking? or are there conditions in which we would worry about performance?

Reply:
I'm still feeling out the patterns that work well, but here are some observations:

The `automock` macro currently can't mock traits with trait bounds, so it doesn't work with Clone. I think it is fine with Send because that is a marker trait that doesn't declare any methods. The more flexible `mock!` macro does support trait bounds, but it requires you to copy-paste each trait into the body of the macro, which is ugly and awkward to maintain.

Removing Clone from the trait doesn't seem like a problem to me, because code that demands a clone-able ledger can always declare additional trait bounds on the types it uses, i.e. `L: Ledger + Clone`.